cmake/KokkosCI.cmake

```diff
@@ -0,0 +1,357 @@
+cmake_minimum_required(VERSION 3.10 FATAL_ERROR)
```

Comment:
Should this also be updated to 3.16?

Reply:
It's not really necessary, you don't run it directly within Kokkos cmake builds